__tests__/app.test.js

```diff
@@ -0,0 +1,68 @@
+const request = require("supertest");
+const app = require("../db/app");
+const db = require("../db/connection");
+const seed = require("../db/seeds/seed");
+const data = require("../db/data/test-data/index");
+
+afterAll(() => {
+  db.end();
+});
+
+beforeEach(() => {
+  return seed(data);
+});
+
+describe("app", () => {
+  describe("/api", () => {
+    it("200: GET - responds with server ok message", () => {
+      return request(app)
+        .get("/api")
+        .expect(200)
+        .then((response) => {
+          expect(response.body.message).toBe("server ok");
+        });
+    });
+  });
+  describe("/api/topics", () => {
+    it("200: GET - responds with an array", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          const { topics } = body;
+          expect(topics).toBeInstanceOf(Array);
+        });
+    });
+  });
+  it("200: GET - responds with an array of all the topic objects, with the correct properties", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            description: expect.any(String),
+            slug: expect.any(String),
+          });
+        });
+      });
+  });
+  it("200: GET - responds with an array of all the topic objects, with the correct data", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        expect(topics).toEqual(data.topicData);
+      });
+  });
+  it("404: should return a status 404 if path is invalid", () => {
```

Comment:
Good test, should probably be in its own describe block though as it isn't exclusive to api/topics. Remove as part of our discussion involving invalid endpoints.